src/content/en/2021/javascript.md

```diff
@@ -1,20 +1,1543 @@
+featured_quote: Features provided for improving the rendering time and resource loading time could be leveraged better to see the overall impact and get an even better experience with respect to the performance.
+featured_stat_1: 4
+featured_stat_label_1: Median asynchronous requests made per mobile page.
+featured_stat_2: 45%
+featured_stat_label_2: Images are the most requested content type on pages.
+featured_stat_3: 2.7%
+featured_stat_label_3: Percent of desktop pages using custom elements.
+unedited: true
 ---
 
-## TODO
+
+### Introduction
+
+The speed and consistency that the JavaScript language has evolved with over the past years is tremendous. While in the past it was  used primarily for client-side, it has taken a very important and respected place in the world of building services and server-side tools. JavaScript has evolved to a point where it is not only possible to create faster applications but also to [run servers within browsers](https://blog.stackblitz.com/posts/introducing-webcontainers/).
+
+There is a lot that happens in the browser when rendering the application, from downloading JavaScript to parsing, compiling and executing it. Let’s start with that first step and try to understand how much JavaScript is actually requested by pages.
+
+
+### How much JavaScript do we load?
+
+To measure is the key towards improvement, they say. To improve the usage of JavaScript in our applications, we need to measure how much JavaScript being shipped is actually required. Let’s dig in to understand the distribution of JavaScript bytes per page, considering what a major role it plays in the web setup.
+
+The 50th percentile (median) desktop page loads 463 kilobytes of JavaScript, whereas a median page load on a mobile device sends 427 kilobytes. 
+
+{{ figure_markup(
+  image="javascript-bytes-per-page.png",
+  caption="Distribution of the amount of JavaScript kilobytes loaded per page.",
+  description="Bar chart showing the distribution of JavaScript bytes per page. The median desktop page loads 463 kilobytes of JavaScript. The 10th, 25th, 50th, 75th, and 90th percentiles for desktop are: 94 KB, 220 KB, 463 KB, 852 KB, and 1,354 KB.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=329775434&format=interactive",
+  sheets_gid="18398250",
+  sql_file="bytes_2021.sql"
+  )
+}}
+
+
+Compared to [2019’s results](https://almanac.httparchive.org/en/2019/javascript#fig-2), this shows an increase of 18.4% in the usage of JavaScript for desktop devices and an increase of 18.9% on mobile devices. As we see the trend is towards using more JavaScript over the years, this could slow down the rendering of an application given the additional CPU work. It's worth to note that these statitics represent the  transferred bytes which could be compressed responses and thus, the actual cost to the CPU could be significantly higher. 
```

Comment:
```suggestion
Compared to [2019’s results](https://almanac.httparchive.org/en/2019/javascript#fig-2), this shows an increase of 18.4% in the usage of JavaScript for desktop devices and an increase of 18.9% on mobile devices. As we see the trend is towards using more JavaScript over the years, this could slow down the rendering of an application given the additional CPU work. It's worth to note that these statistics represent the transferred bytes which could be compressed responses and thus, the actual cost to the CPU could be significantly higher. 
```